econml/iv/dml/_dml.py

```diff
@@ -761,6 +761,14 @@ def predict(self, Y, T, X=None, W=None, Z=None, sample_weight=None, groups=None)
             TX_pred = np.tile(TX_pred.reshape(1, -1), (T.shape[0], 1))
         Y_res = Y - Y_pred.reshape(Y.shape)
         T_res = TXZ_pred.reshape(T.shape) - TX_pred.reshape(T.shape)
+        if T_res.sum() == 0:
+            raise ValueError(
+                """
+                All values of the treatment residual are 0, 
+                which then makes them unsuitable to use as weights 
+                in downstream in econml/dml/dml.py 
```

Comment:
Linting is failing because these lines all end in trailing whitespace.  I'd tweak the last part of the message slightly:

```suggestion
                All values of the treatment residual are 0,
                which then makes them unsuitable to use as weights.
                DRIV requires that the instrument Z has an effect on the
                expected treatment value of at least some rows.

                If you are using regularized models, it's possible that this error is a
                result of regularizing too strongly, so that all predictions from both
                models are constant.
```